src/riak_core_info_service_process.erl

```diff
@@ -0,0 +1,69 @@
+-module(riak_core_info_service_process).
+-behaviour(gen_server).
+
+-record(state, {
+    source = undefined :: riak_core_info_service:callback(),
```

Comment:
+1 - makes sense to me... will think about the name a bit but I like `service_provider`